src/hooks/useAxios.js

```diff
@@ -0,0 +1,38 @@
+import axios from "axios";
+import { useState, useCallback } from "react";
+/**
+ * A custom hook used to fetch data and from the API
+ * @param {string} url  Address of the API we connect to fetch data
+ * @returns Boolean that shows loading state, error information and a function that fetches the data with given url
+ */
+const useAxios = (url) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setIsError] = useState(null);
+
+  const fetchData = useCallback(
+    async (setData) => {
+      setIsLoading(true);
+      try {
+        const response = await axios.get(url);
```

Comment:
You can destructure here like:
const { status, data }= await axios.get(url);

And after you don't need line 20 at all/